src/env/node/git/git.ts

```diff
@@ -160,6 +166,15 @@ function getStdinUniqueKey(): number {
 type ExitCodeOnlyGitCommandOptions = GitCommandOptions & { exitCodeOnly: true };
 export type PushForceOptions = { withLease: true; ifIncludes?: boolean } | { withLease: false; ifIncludes?: never };
 
+const branchErrorAndReason = [
+	[GitErrors.noRemoteReference, BranchErrorReason.NoRemoteReference],
+	[GitErrors.invalidBranchName, BranchErrorReason.InvalidBranchName],
+	[GitErrors.branchAlreadyExists, BranchErrorReason.BranchAlreadyExists],
+	[GitErrors.branchNotFullyMerged, BranchErrorReason.BranchNotFullyMerged],
+	[GitErrors.branchNotYetBorn, BranchErrorReason.BranchNotYetBorn],
+	[GitErrors.branchFastForwardRejected, BranchErrorReason.BranchFastForwardRejected],
```

Comment:
These don't exist yet - you'll need to define them. For my own curiosity, what is "branch not yet born?"